lte/gateway/python/magma/pipelined/app/enforcement_stats.py

```diff
@@ -599,21 +635,50 @@ def get_stats(self, cookie: int = 0, cookie_mask: int = 0):
             self.logger.error("Could not obtain rule records due to either InvalidDatapath, OFError or UnexpectedMultiReply")
             return RuleRecordTable()
 
-def _generate_rule_match(imsi, ip_addr, rule_num, version, direction):
+    def _prepare_session_config_report(self, stats_msgs):
+        session_config_dict={}
+
+        for flow_stats in stats_msgs:
+            for stat in flow_stats:
+                if stat.table_id != self.tbl_num:
+                    return
+
+                local_f_teid_ng = _get_ng_local_f_id(stat)
+                if not local_f_teid_ng or local_f_teid_ng == REG_ZERO_VAL:
+                    continue
+                #Already present
+                if local_f_teid_ng in session_config_dict:
+                    continue
+
+                sid = _get_sid(stat)
+                if not sid:
+                    continue
+
+                rule_version = _get_version(stat)
+                session_config_dict.update ({local_f_teid_ng:
```

Comment:
this seems like a weird way of doing
session_config_dict[local_f_teid_ng]=UPFSessionState(subscriber_id=sid, session_version=rule_version, local_f_teid=local_f_teid_ng)

Reply:
Yes. Made changes